cimgen/languages/modernpython/utils/base.py

```diff
@@ -132,38 +151,217 @@ def cgmes_attributes_in_profile(self, profile: BaseProfile | None) -> dict[str,
-                        # The attribute has some extras, but not namespace.
-                        # Use the class namespace.
-                        namespace = self.namespace
-                    else:
+
+                # Namespace finding
+                # "class namespace" means the first namespace defined in the inheritance tree.
+                # This can go up to Base, which will give the default cim NS.
+                infos["namespace"] = self.namespace
+                extra = getattr(f.default, "json_schema_extra", None)
+                if extra and extra.get("is_used"):
+                    # adding the extras, used for xml generation
+                    extra_info = {
+                        "attr_name": qualname,
+                        "is_class_attribute": extra.get("is_class_attribute"),
+                        "is_enum_attribute": extra.get("is_enum_attribute"),
+                        "is_list_attribute": extra.get("is_list_attribute"),
+                        "is_primitive_attribute": extra.get("is_primitive_attribute"),
+                        "is_datatype_attribute": extra.get("is_datatype_attribute"),
+                        "attribute_class": extra.get("attribute_class"),
+                        "attribute_main_profile": self.get_attribute_main_profile(shortname),
+                    }
+                    if extra.get("namespace"):
                         # The attribute has an explicit namesapce
-                        namespace = attr_ns
+                        extra_info["namespace"] = extra.get("namespace", self.namespace)
+                    infos.update(extra_info)
 
-                    qual_attrs[qualname] = CgmesAttribute(
-                        value=getattr(self, shortname),
-                        namespace=namespace,
-                    )
-                    seen_attrs.add(shortname)
+                infos["value"] = getattr(self, shortname)
+
+                qual_attrs[qualname] = CgmesAttribute(infos)
+                seen_attrs.add(shortname)
 
         return qual_attrs
 
+    def to_xml(self, profile_to_export: BaseProfile, id: Optional[str] = None) -> Optional[etree.Element]:
+        """Creates an etree element of self with all non-empty attributes of the profile_to_export
+        that are not already defined in the recommanded profile
+        This can then be used to generate the xml file of the profile_to_export
+        Args:
+            profile_to_export (Profile): Profile for which we want to obtain the xml tree (eg. Profile.EQ)
+            id (Optional[str], optional): "mRID", optional: some objects don't have mRID attribute. Defaults to None.
+        Returns:
+            Optional[etree.Element]: etree describing self for the profile_to_export, None if nothing to export
+        """
+        profile_attributes = self._get_attributes_to_export(profile_to_export)
+
+        if "mRID" in self.to_dict():
+            obj_id = self.mRID
```

Comment:
```suggestion
            obj_id = self.mRID  # type: ignore
```
Maybe "assert self.mRID" or similar would be better?